serial_protocol.md

```diff
@@ -186,3 +192,135 @@ Used to stop listening to events at a given path. This must be the same path pro
 	>>BEGIN_STREAM /user/aturing
 	>>END_STREAM /user/aturing
 	<<+OK
+
+#Messaging Protocol
+
+## BEGIN_MESSAGING 
+Called to start communicating with Firebase Cloud Messaging, it requires a server key.
+###Usage
+	BEGIN %Server_Key%
+###Response
+	+OK - Accepted initialization parameters
+###Examples
+	>> BEGIN https://samplechat.firebaseio-demo.com
+	<< +OK
+
+## MESSAGE_TO_USERS
+Called to start composing a message to users by registration id. You can specify as many as you wish separated by spaces on the same line. 
```

Comment:
Sorry didn't actually see your other comment before my last comment, I guess github wasn't updating my page.

So even if we eliminate the arguments, which I am not sure I agree with, than we are still left with the problem of multiline data and notifications which is surely going to be wanted. I am still pretty against requiring people to send json. Asking them to escape also seems problematic. What is the objection to the multi-command message building? I guess there is always danger that the client and the firething could become out of sync. So the firething still thinks it is in a message but the client has restarted and is trying to send it other stuff. We could implement some logic like we abort the message on the first failure. That may be desirable anyway, if the client code sends something bad it is unlikely to have any way to fix it, only a user would. This would also match the way SMTP and similar protocols work.